src/Repo_new.java

```diff
@@ -0,0 +1,804 @@
+            path.insert(0, '/').insert(0, f.getName());
+        }
+
+        return path.toString().intern();
+    }
+
+    public static String quotePath(final String path) {
+        if (path.indexOf(' ') != -1) {
+            return '"' + path + '"';
+        } else {
+            return path;
+        }
+    }
+
+    public static String unquotePath(final String path) {
+        final int len = path.length();
+
+        if ((len > 2) && (path.charAt(0) == '"')
+                && (path.charAt(len - 1) == '"')) {
+            return path.substring(1, len - 1);
+        } else {
+            return path;
+        }
+    }
+
+    public String toCommitter(final int userId) {
+        String name = getRepoProperties().getUserMapping(userId);
+
+        if (name != null) {
+            return name;
+        }
+
+        final View v = getView();
+
+        if (v != null) {
+            final User user = getView().getServer().getUser(userId);
+
+            if (user != null) {
+                name = user.getName();
+            }
+        }
+        if (name == null) {
+            name = "Unknown User";
+        }
+
+        final List<String> l = new ArrayList<>();
+        l.add(name);
+
+        for (final StringTokenizer st = new StringTokenizer(name, " ,"); st
+                .hasMoreTokens();) {
+            final String next = st.nextToken();
+            l.add(next);
+            l.add(next.toLowerCase());
+        }
+
+        name = MessageFormat.format(getUserNamePattern(), l.toArray());
+        getRepoProperties().setSessionUserMapping(userId, name);
+        return name;
+    }
+
+    private static Project findProject(final Server s, final String name) {
+        for (final Project p : s.getProjects()) {
+            if (name.equals(p.getName())) {
+                return p;
+            }
+        }
+
+        throw new ConfigurationException("No such project: " + name);
+    }
+
+    private static View findView(final Project p, final String name) {
+        for (final View v : p.getViews()) {
+            if (name.equals(v.getName())) {
+                return v;
+            }
+        }
+
+        throw new ConfigurationException("No such view: " + name);
+    }
+
+    private static String getFileName(final String path) {
+        final int slash = path.lastIndexOf('/');
+
+        if (slash == -1) {
+            return path;
+        } else if (slash == (path.length() - 1)) {
+            return "";
+        } else {
+            return path.substring(slash + 1);
+        }
+    }
+
+    private Folder getOrCreateParentFolder(final String path) {
+        final int slash = path.lastIndexOf('/');
+
+        if (slash == -1) {
+            return getRootFolder();
+        } else {
+            return getOrCreateFolder(path.substring(0, slash));
+        }
+    }
+
+    private static File createFile(final File dir, final String path) {
+        File f = new File(dir, path);
+        for (int i = 1; f.exists(); i++) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This loop's stop condition tests "f" but the incrementer updates "i". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=squid%3AS1994)